recipes/imath/all/conanfile.py

```diff
@@ -45,6 +47,17 @@ def validate(self):
         if self.settings.compiler.get_safe("cppstd"):
             check_min_cppstd(self, 11)
 
+        # Check for unsupported gcc 7 with version 3.1.10
+        compiler = self.settings.compiler
+        compiler_name = str(compiler)
+        is_invalid_gcc = (compiler_name == 'gcc' and Version(compiler.version) < '7')
+        is_version_3_1_10 = (Version(self.version) == '3.1.10')
+        if is_invalid_gcc and is_version_3_1_10:
+            raise ConanInvalidConfiguration(
+                f"Minimum required gcc version: 7 to include infinite in <cmath> with imath 3.1.10."
+            )
+
```

Comment:
What about backporting https://github.com/AcademySoftwareFoundation/Imath/pull/371 and https://github.com/AcademySoftwareFoundation/Imath/pull/368 to enable old gcc behaviour? I know that supporting such an old gcc release is quite tedious, but turns out it's still a popular version :$

Reply:
I can do the patching for you if needed :)

Reply:
Thanks for offering. I just added a patch derived from the two changes.